controllers/discovery/initDiscoveryUserData.js

```diff
@@ -69,11 +75,23 @@ const InitDiscoveryUserData = async (req, res) => {
     });
     let suggestedUsers = usersWithCommonFollowerResult;
 
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
+    totalData = totalData?.[0]?.total_data || 0;
+
     return res.status(200).json({
       success: true,
       message: `Fetch discovery data success`,
       suggestedUsers,
-      nextPage: page + 1
+      page: page + 1,
+      total_page: totalData > 0 && limit > 0 ? Math.ceil(totalData / limit) : 0,
+      limit: limit,
+      offset: page * limit
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The response object is well structured and provides useful information for pagination. However, the `offset` field seems redundant as it can be calculated from `page` and `limit`. Removing this field would simplify the response.
```diff
-       offset: page * limit
```
Also, consider adding error handling to send a proper response in case of an exception. Currently, the catch block only logs the error but does not send a response back to the client.
```diff
-     console.log('e');
+     res.status(500).json({ error: 'An error occurred while fetching discovery data' });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->